### PR TITLE
go/cmd/registry: simplify register runtime command

### PR DESCRIPTION
**Pull request by ptrus**
Closes: https://github.com/oasisprotocol/oasis-core/issues/3626

Update `gen_register` runtime command to accept runtime descriptor instead of messing with the arguments.

TODO:
- [x] update deploying a runtime docs

### CODE REVIEW

File: .changelog/3903.internal.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+go/cmd/registry: simplify register runtime command
```
Comment (kostko) [inline .changelog/3903.internal.md:1]: I don't think this is internal, but rather a "feature" (or even breaking, but we'll want to backport this).

File: docs/setup/deploying-a-runtime.md [hunk 2]
```diff
@@ -66,51 +70,102 @@ network.
...
 To generate and sign a runtime registration transaction that will initialize and
 register the runtime we will use the `registry runtime gen_register` command.
-When initializing a runtime we need to specify various runtime parameters. To
-list all of the available parameters with short descriptions run:
-`registry runtime gen_register --help` subcommand.
-
-```
-oasis-node registry runtime gen_register --help
-```
+When initializing a runtime we need provide the runtime descriptor.
```
Comment (kostko) [inline docs/setup/deploying-a-runtime.md:73]:
```suggestion
When initializing a runtime we need to provide the runtime descriptor.
```

File: .changelog/3903.internal.md (hunk 1 above)
Comment (tjanez) [reply .changelog/3903.internal.md]: I think this should be classified as breaking.

File: .changelog/3903.feature.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+go/cmd/registry: simplify register runtime command
+
+The `registry runtime gen_register` command now accepts a JSON runtime
+descriptor, which should simplify generating runtime registration
+transactions.
+The `registy runtime init_genesis` command is removed as runtime
+descriptors in genesis are not singed since `v21.x` making the command
```
Comment (tjanez) [inline .changelog/3903.feature.md:7]:
```suggestion
descriptors in genesis are not singed since version 21.0+ making the command
```

**Review by kostko — APPROVED** [review]

**Pull request merged**